docs/src/implemented-proposals/multi-level-multi-iterator-scheduler.md

```diff
@@ -0,0 +1,84 @@
+
+## Single-Level Multi-Iterator
+
+The single-level multi-iterator has already been implemented in Solana's independent banking stage thread schedulers, this section serves to outline that concept in order to provide context for the multi-level multi-iterator.
+
+First, we take our priority-queue and collect the entire buffer into a vector. We initialize multiple, in this case 64, positions such that there are no write-conflicts between the transactions at the given positions. These, up to 64, transactions then make up a transaction batch to be executed. The multi-iterator marks these transactions as `already_processed`, and the positions are marched forward, again checking for conflicts as we set them.
+
+### Example
+
+For simplicity's sake, let's assume we have a batch-size of 2 instead of 64, and a priority-queue with the following transactions:
+
+```text
+[A, A, A, B, C, A, B, C, D]
+ ^        ^                     // initial iterator positions
+```
+
+The first batch would be `[A, B]`. Marching the positions forward, we get:
+
+```text
+[A, A, A, B, C, A, B, C, D]
+    ^        ^
+```
+
+and the second batch is `[C, D]`. Marching the positions forward again, we get:
+
+```text
+[A, A, A, B, C, A, B, C, D]
+       ^           ^
+```
+
+giving a batch of `[A, B]`. Note that the second iterator skipped over an `A` transaction, as that transaction would conflict with the `A` our first position is pointing to. Following this the remaining batches would be `[[A, C], [D]]`.
+
+## Multi-Level Multi-Iterator
+
+The multi-iterator works well for a single thread, but we should not use it directly in a centralized scheduler. If we have two independent hot events touching accounts `A` and `B`, the single-level multi-iterator would put them into the same batch, which would then be serialized. This is not ideal, as we would like to be able to execute these transactions in parallel.
+
+The multi-level multi-iterator uses multiple multi-iterators to construct the batches for our threads, while maintaining that all transactions can be executed in parallel, and that the hottest events would get scheduled to separate threads.
+
+If we have 4 threads and a batch-size of 64, the multi-level multi-iterator would have 64 multi-iterators each with 4 iterators. Batches for the for the first thread would be constructed by taking the transactions pointed to by the first-iterator of each multi-iterator, the second thread would take the second-iterator of each multi-iterator, and so on. At each step, the first multi-iterator marches all its' positions forward, then the second multi-iterator marches its' positions forward, and so on.
```

Comment:
"for the for the first" -> "for the"

Reply:
Might also be good to state what happens if there are more/less threads than iterators, or do we assume they will be equal?

Reply:
Fixed the typo.

For the more/less threads, see the example below where we have 4 iterators and 2 threads.